build/dockerfiles/rhel.entrypoint.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+#
+# Copyright (c) 2012-2018 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+if ! whoami &> /dev/null; then
+  if [ -w /etc/passwd ]; then
+    echo "${USER_NAME:-jboss}:x:$(id -u):0:${USER_NAME:-jboss} user:${HOME}:/sbin/nologin" >> /etc/passwd
```

Comment:
we only need this in RHEL, not alpine/ubuntu?

Reply:
The registry has been working fine without it since its inception; unless there's a compelling case for including it without knowing why, I'd rather leave it out.

I'm against pushing rhel-related complexity into the default image for the sake of consistency.

Reply:
yeah turns out this was a mistake... the container runs as root so this jboss user thing is not needed.

Reply:
It may still be needed, but I'd rather add it when we know it's necessary. On OpenShift, the way things work is that containers are started running as a (mostly random) UID. As far as I know, the things that are impacted by this are:
- File permissions: you basically can't rely user-level permissions in container files, since the UID is inconsistent across deployments; any files that you need read/write/execute need to be owned by the root group and have those permissions there
- Username dependency: Since your UID is generated at startup, you won't have a username or home directory set up; this can cause problems for userspace programs that rely on e.g. storing things in the home directory, or anything that depends on being able to access a username. I don't *think* this impacts Apache, since it doesn't touch user files. The containers we're using already have appropriate permissions for everything Apache needs. 

The second part is what the line above is designed to address (so that a username is available). We need it on Che workspace images, since there we're basically emulating a user terminal/IDE and various apps there depend on a username, home directory, login shell, etc.

FWIW also, I believe OpenShift 4.x clusters can be configured to automatically provision an entry in `/etc/passwd` at container startup using cri-o, but I haven't played with it too much.